projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py

```diff
@@ -46,102 +54,174 @@ def _execute(self, command):
 
     @LogDecorator(log)
     def login(self):
-        self._execute(
-            [
-                "login",
-                "-u",
-                f"{self.config.vdkcli_oauth2_uri}",
-                "-a",
-                f"{self.config.vdkcli_api_refresh_token}",
-                "-t",
-                "api-token",
-            ]
-        )
+        if self.config.vdkcli_oauth2_uri:
+            self._execute(
+                [
+                    "login",
+                    "-u",
+                    f"{self.config.vdkcli_oauth2_uri}",
+                    "-a",
+                    f"{self.config.vdkcli_api_refresh_token}",
+                    "-t",
+                    "api-token",
+                ]
+            )
+        else:
+            self._execute(
+                [
+                    "login",
+                    "-a",
+                    f"{self.config.vdkcli_api_refresh_token}",
+                    "-t",
+                    "api-token",
+                ]
+            )
```

Comment:
```

    def __control_api_url_arg(self):
        if self.config.vdkcli_oauth2_uri:
            return ["-u", f"{self.config.vdkcli_oauth2_uri}"]
        else 
            return []

    @LogDecorator(log)
    def login(self):
        self._execute(["login"] + ["-t", "api-token"] + self.__control_api_url_arguments())
```

Perhaps something akin to above would be slightly simpler and clearer. Here in those arrays people would be playing a bit [Spot the difference](https://en.wikipedia.org/wiki/Spot_the_difference) :)

Reply:
Yes, thank you! I spent some time trying to find a way to do that, but everything I could think of, was awful :D